### PR TITLE
Migrate p-any and p-all to native functions

### PR DESCRIPTION
**Pull request by marcusmotill**
## Description
- removed dependency on p-any and p-all
- remove `start` function in favor of `some` (alias to `any`)

## How Has This Been Tested
- added new tests for added functions at 100% coverage
- fixed bad reporting of limiter test

### CODE REVIEW

File: test/Limiter.js [hunk 1]
```diff
@@ -7,166 +7,196 @@ const assert = require('assert');
 describe('Limiter test', function() {
     this.timeout(30000);
 
-    it('should return if no collection is supplied', done => {
-        const limiter = new the.Limiter();
+    it('should return if no collection is supplied', async () => {
```
Comment (marcusmotill) [inline test/Limiter.js]: needed to wrap all these tests in a promise and reject bad results or else mocha would not report the test as failed

File: test/any.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        try {
+            await the.any(collection, task);
+        } catch (e) {
+            err = e;
+        }
+        assert(err);
+        assert(err instanceof Error);
+    });
+
+    it('should not error if last item resolves (edge case)', async () => {
```
[the remark this answers is not in the document]
Comment (marcusmotill) [reply test/any.js]: last item resolves and 'done' message beats the iteration message
